common/utilities/string/wrap.cpp

```diff
@@ -0,0 +1,130 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+#include <string>
+#include <vector>
+#include <sstream>
+#include "wrap.h"
+#include "split.h"
+#include "../third-party/imgui/imgui.h"
+
+namespace utilities {
+namespace string {
+
+std::string wrap_paragraph( const std::string & paragraph, int wrap_pixels_width )
+{
+    float space_width = ImGui::CalcTextSize( " " ).x;  // Calculate space width in pixels
+    std::string wrapped_line;                          // The line that is wrapped in this iteration
+    std::string wrapped_paragraph;                     // The output wrapped paragraph
+    auto remaining_paragraph = paragraph;              // Holds the remaining unwrapped part of the input paragraph
+
+    // Handle a case when the paragraph starts with spaces
+    if( remaining_paragraph[0] == ' ' )
+    {
+        auto non_space_index = remaining_paragraph.find_first_not_of( ' ' );
+        // If non spaces characters exist
+        if( non_space_index != std::string::npos )
+        {
+            // Remove trailing spaces
+            remaining_paragraph
+                = remaining_paragraph.substr( non_space_index );
+        }
+    }
+
+    auto next_word = remaining_paragraph.substr(
+        0,
+        remaining_paragraph.find( ' ' ) );  // The next word to add to the current line
+    bool first_word = true;
+
+    while( ! next_word.empty() )
+    {
+
+        float next_x = 0.0f;
+        // If this is the first word we try to place it first in line,
+        // if not we concatenate it to the last word after adding a space
+        if( ! first_word )
+        {
+            next_x = ImGui::CalcTextSize( wrapped_line.c_str() ).x + space_width;
+        }
+
+        if( next_x + ImGui::CalcTextSize( next_word.c_str() ).x <= wrap_pixels_width )
+        {
+            if( ! first_word )
+                wrapped_line += " ";  // First word should not start with " "
```

Comment:
So if I `wrap( "ab  cd" )` it will give the same result as `wrap( "ab cd" )`?

Reply:
Yes.
I started to find a way to handle it but it adds lots of code.
I suggest adding it to the header function description as a known behavior , multiple spaces that need to be kept is a rare corner case that I don't think we will meet. 
What do you think?

Reply:
I don't like it.
It's acceptable to commit with known limitation, but then it needs to be documented and made part of the unit-test with an explanation.

Reply:
And I see you've done exactly that :)